server/lib/session/session-api.ts

```diff
@@ -148,6 +151,15 @@ export class SessionApi {
     const perms = (await getPermissions(user.id))!
     await maybeMigrateSignupIp(user.id, ctx.ip)
 
+    if (language && language !== user.language) {
```

Comment:
No, don't do this. If a user has set a specific language, that setting always takes precedent over whatever is auto-detected, period. Please think through user flows when deciding stuff like this:

- User logs in, sees the language in Spanish but they want English
- User goes to settings, picks English
- User logs out for whatever reason
- User opens app later, Electron picks Spanish again
- User logs in

 With your code, their setting is now Spanish again, not good.

The only time you should be updating this column is:

1) When a user logs in or retrieves a current session, and they had no detected language stored
2) When the user explicitly changes their language (via settings)

The language parameter here is not a "please change my account's language" directive, it is "here is a language to use to migrate my account if necessary"